stetho/src/main/java/com/facebook/stetho/inspector/elements/Descriptor.java

```diff
@@ -55,6 +57,41 @@ public final void postAndWait(Runnable r) {
     getHost().postAndWait(r);
   }
 
+  /**
+   * Parses the text argument text from DOM.setAttributeAsText()
+   * Text will be in the format "attribute1=\"Value 1\" attribute2=\"Value2\""
+   * @param text the text argument to be parsed
+   * @return a map of attributes to their respective values to be set.
+   */
+  protected Map<String, String> parseSetAttributesAsTextArg(String text) {
+    char[] chars = text.toCharArray();
+    String value = "";
+    String key = "";
+    String buffer = "";
```

Comment:
we should use a StringBuilder for this. building a String by appending to it in a loop is a classic performance anti-pattern in languages like Java and C#